sdks/python/apache_beam/io/gcp/experimental/bigtableio_read_it_test.py

```diff
@@ -0,0 +1,169 @@
+from apache_beam.runners.runner import PipelineState
+
+try:
+  from google.cloud.bigtable import Client
+except ImportError:
+  Client = None
+
+
+@unittest.skipIf(Client is None, 'GC Bigtable dependencies are not installed')
+class BigtableReadTest(unittest.TestCase):
+  """ Bigtable Read Connector Test
+
+  This tests the ReadFromBigtable connector class via reading rows from
+  a Bigtable table and comparing the `Rows Read` metrics with the row
+  count known a priori.
+  """
+  def setUp(self):
+    self.options = parse_commane_line_arguments()
+
+    logging.info('\nProject ID:  %s', self.options['project'])
+    logging.info('\nInstance ID: %s', self.options['instance'])
+    logging.info('\nTable ID:    %s', self.options['table'])
+
+    self._p_options = PipelineOptions(**self.options)
+    self._p_options.view_as(SetupOptions).save_main_session = True
+
+    logging.getLogger().setLevel(self.options['log_level'])
+
+    # [OPTIONAL] Uncomment to save logs into a file
+    # self._setup_log_file()
+
+    # [OPTIONAL] Uncomment this to allow logging the pipeline options
+    # for key, value in self.p_options.get_all_options().items():
+    #   logging.info('Pipeline option {:32s} : {}'.format(key, value))
+
+  def _setup_log_file(self):
+    if self.options['log_directory']:
+      # logging.basicConfig(
+      #   filename='{}{}.log'.format(
+      #       options['log_directory'], options['table']
+      #   ),
+      #   filemode='w',
+      #   level=logging.DEBUG
+      # )
+
+      # Forward all the logs to a file
+      fh = logging.FileHandler(
+          filename='{}test_log_{}_RUNNER={}.log'.format(
+              self.options['log_directory'],
+              self.options['table'][-15:],
+              self.options['runner']))
+      fh.setLevel(logging.DEBUG)
+      logging.getLogger().addHandler(fh)
+
+  @attr('IT')
+  def test_bigtable_read(self):
+    logging.info(
+        'Reading table "%s" of %d rows...',
+        self.options['table'],
+        self.options['row_count'] or 0)
+
+    p = beam.Pipeline(options=self._p_options)
+    _ = (
+        p | 'Read Test' >> ReadFromBigtable(
+            project_id=self.options['project'],
+            instance_id=self.options['instance'],
+            table_id=self.options['table'],
+            filter_=self.options['filter']))
+    self.result = p.run()
+    self.result.wait_until_finish()
+    assert self.result.state == PipelineState.DONE
+
+    query_result = self.result.metrics().query(
+        MetricsFilter().with_name('Rows Read'))
+
+    if query_result['counters']:
+      read_counter = query_result['counters'][0]
+      final_count = read_counter.committed
+      assert final_count == self.options['row_count']
+      logging.info(
+          '%d out of %d rows were read successfully.',
+          final_count,
+          self.options['row_count'])
+
+    logging.info('DONE!')
+
+
+def parse_commane_line_arguments():
+  parser = argparse.ArgumentParser()
+
```

Comment:
You do not need to set these up manually. Please see the way our other integration tests are setup. You can either configure a new Jenkins test suite or add integration tests to existing Python post-commit tests. Latter might be easier.